server.go

```diff
@@ -186,6 +235,248 @@ func main() {
 	}
 }
 
+const msgUpstreamParticipantCreateError = "could not create upstream participant. response status: %s"
```

Comment:
You COULD create a type that implements the interface for `error` and control the output that way?

```
type NotFoundError struct {
    Name string
}

func (e *NotFoundError) Error() string { return e.Name + ": not found" }

if e, ok := err.(*NotFoundError); ok {
    // e.Name wasn't found
}
```
Something akin to that (stolen from: https://blog.golang.org/go1.13-errors)

Reply:
tried 'error types' here: a751930
please holler if it could be done better